summary.Rmd

```diff
@@ -0,0 +1,41 @@
+Summary
+=======
+
+1. Use available tools
+    a. Image registries (e.g. Docker Hub),
+    b. Dockerfile generators (e.g. repo2docker)
+    c. Project templates (e.g. [cookie cutter](https://github.com/cookiecutter/cookiecutter))
+    d. Linters
+2. Use versioned images
+    a. Mention specific sha digests?
+3. Format for clarity
+    a. Use indentation, line breaks and unabbreviated arguments
+    b. Clarity > brevity
+    c. Group related commands
+4. Document within the Dockerfile
+    a. Add comments explaining the steps
+    b. Use labels for metadata
+    c. Include build and run commands
+6. Specify software versions
+    a. Pin system library versions
+    b. Explictly list package versions to install
+    c. Comment why each is needed
+9. Use version control
+    a. Use Continuous Integration and automated testing
+    b. ~~Use one Dockerfile per project~~
```

Comment:
This is a good spot for point b (one Dockerfile per project, since the reader is thinking about the structure of the version controlled repository). I'm still thinking the cookie cutter templates bit belongs best where we have it (point 1) so I'd remove it here.

Reply:
These are crossed out because templating was moved to Rule 1 and "one Dockerfile per project" was dropped for the more general statement.

Reply:
I'm :+1: for moving the cookie cutter templates to rule #1.

Reply:
Great!